src/lib/history.ts

```diff
@@ -0,0 +1,29 @@
+import { WeeklyArticleLog, WeeklyArticleHistory } from '../components/ContributionsEpicTypes';
+
+export const getMondayFromDate = (date: Date): number => {
+    const day = date.getDay() || 7;
+    if (day !== 1) {
+        date.setHours(-24 * (day - 1));
+    }
+    return Math.floor(date.getTime() / 86400000);
+};
+
+export const getArticleViewCountForWeeks = (
+    history: WeeklyArticleHistory = [],
+    weeks: number,
+): number => {
+    const mondayThisWeek = getMondayFromDate(new Date());
```

Comment:
Oh I forgot to ask before. We're relying on `new Date()` here - do we need to mock that out somehow so that these tests don't start failing down the line?

Reply:
I've refactored the `getArticleViewCountForWeeks` function to accept an optional 3rd argument with the date to do the checks against. This will be initialised as `new Date()` so the current use cases remain unchanged, but the tests will use that argument to lock in the date being checked against.